packages/api/src/controllers/customAnimalTypeController.js

```diff
@@ -39,7 +39,8 @@ const customAnimalTypeController = {
       const trx = await transaction.start(Model.knex());
       try {
         const { farm_id } = req.headers;
-        const { type } = req.body;
+        let { type } = req.body;
+        type = type.trim();
```

Comment:
Oh you also want to enforce it's not empty? Is that what you're doing for the other POST controller?

Reply:
@Duncan-Brain do you prefer it abstracted? Would you want to use it your controller?

```js
  /**
   * Checks if the input is a string and not just white space.
   * Returns a trimmed version of the input if it's a valid string, or null
   *
   * @param {string} input - The input string to validate and trim.
   * @return {string | null} - The trimmed string if valid, otherwise null.
   */
  checkAndTrimString(input) {
    if (typeof input !== 'string' || !input.trim()) {
      return null;
    }
    return input.trim();
  },
  ```

Reply:
I am good with any way really haha! Just asking that we probably don't want empty strings in the db is all I am thinking. I havent gotten to that part in the post yet I decided to review others PRs as first priority.

If I were to really really abstract it haha... it would probably take an array of objects to check the null status of and handle the the rollback and error. But thats probably overkill -- the null or empty string check is quite fine and very clear.